org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/participants/codeactions/cvc_complex_type_2_4_aCodeAction.java

```diff
@@ -0,0 +1,180 @@
+package org.eclipse.lsp4xml.extensions.contentmodel.participants.codeactions;
```

Comment:
I think it's a bad idea to use diagnostic message to collect allowed elements because message could be formatted as markdown or other kind. 

The idea is to retrieve those allowed elements with the content model manager (used for completion)

Here the code that I wrote but I have not tested when element uses prefix:

```java
package org.eclipse.lsp4xml.extensions.contentmodel.participants.codeactions;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

import org.eclipse.lsp4j.CodeAction;
import org.eclipse.lsp4j.Diagnostic;
import org.eclipse.lsp4j.Range;
import org.eclipse.lsp4xml.commons.CodeActionFactory;
import org.eclipse.lsp4xml.dom.DOMDocument;
import org.eclipse.lsp4xml.dom.DOMElement;
import org.eclipse.lsp4xml.dom.DOMNode;
import org.eclipse.lsp4xml.extensions.contentmodel.model.CMElementDeclaration;
import org.eclipse.lsp4xml.extensions.contentmodel.model.ContentModelManager;
import org.eclipse.lsp4xml.services.extensions.ICodeActionParticipant;
import org.eclipse.lsp4xml.services.extensions.IComponentProvider;
import org.eclipse.lsp4xml.settings.XMLFormattingOptions;
import org.eclipse.lsp4xml.utils.LevenshteinDistance;
import org.eclipse.lsp4xml.utils.XMLPositionUtility;

/**
 * cvc_complex_type_2_4_a
 */
public class cvc_complex_type_2_4_aCodeAction implements ICodeActionParticipant {

	private static final float MAX_DISTANCE_DIFF_RATIO = 0.4f;

	@Override
	public void doCodeAction(Diagnostic diagnostic, Range range, DOMDocument document, List<CodeAction> codeActions,
			XMLFormattingOptions formattingSettings, IComponentProvider componentProvider) {
		try {
			int offset = document.offsetAt(diagnostic.getRange().getStart());
			DOMNode node = document.findNodeAt(offset);
			if (node != null && node.isElement()) {
				// Get element from the diagnostic
				DOMElement element = (DOMElement) node;
				DOMElement parentElement = element.getParentElement();
				String tagName = element.getTagName();

				// Get the XSD element declaration of the parent element.
				ContentModelManager contentModelManager = componentProvider.getComponent(ContentModelManager.class);
				CMElementDeclaration cmElement = contentModelManager.findCMElement(parentElement);
				if (cmElement != null) {
					// Collect all possible elements from the parent element upon the offset start
					// of the element
					Collection<CMElementDeclaration> possibleElements = cmElement.getPossibleElements(parentElement,
							element.getStart());
					List<CMElementDeclaration> similarElements = new ArrayList<>();
					// Try to collect similar names coming from tag name
					for (CMElementDeclaration possibleElement : possibleElements) {
						if (isSimilar(possibleElement.getName(), tagName)) {
							similarElements.add(possibleElement);
						}
					}

					// Create ranges for the replace.
					List<Range> ranges = new ArrayList<>();
					ranges.add(XMLPositionUtility.selectStartTag(element));
					Range end = XMLPositionUtility.selectEndTag(element);
					if (end != null) {
						ranges.add(end);
					}

					if (!similarElements.isEmpty()) {
						// // Add code actions for each similar elements
						for (CMElementDeclaration similarElement : similarElements) {
							String elementName = similarElement.getName();
							CodeAction similarCodeAction = CodeActionFactory.replaceAt(
									"Did you mean '" + elementName + "'?", elementName, document.getTextDocument(),
									diagnostic, ranges);
							codeActions.add(similarCodeAction);
						}
					} else {
						// Add code actions for each possible elements
						for (CMElementDeclaration possibleElement : possibleElements) {
							String elementName = possibleElement.getName();
							CodeAction otherCodeAction = CodeActionFactory.replaceAt(
									"Replace with '" + elementName + "'", elementName, document.getTextDocument(),
									diagnostic, ranges);
							codeActions.add(otherCodeAction);
						}
					}
				}
			}

		} catch (Exception e) {
			// Do nothing
		}
	}

	private static boolean isSimilar(String reference, String current) {
		int threshold = Math.round(MAX_DISTANCE_DIFF_RATIO * reference.length());
		LevenshteinDistance levenshteinDistance = new LevenshteinDistance(threshold);
		return levenshteinDistance.apply(reference, current) != -1;
	}
}
```